cms/djangoapps/appsembler/views.py

```diff
@@ -274,3 +281,39 @@ def log_multiple_objects_returned(self):
     def render_login_page_with_error(self, error_code):
         return render_login_page(
             login_error_message=self.error_messages[error_code])
+
+
+def get_logout_redirect_url(request):
+    """
+    Return logout redirect url using the site related to the given user if possible.
+    Otherwise, return settings.LOGOUT_REDIRECT_URL
+
+    :return: logout redirect url or settings.LOGOUT_REDIRECT_URL
+    """
+    user = getattr(request, 'user', None)
+    if not user or not user.is_authenticated:
+        return reverse(settings.LOGOUT_REDIRECT_URL)
+
+    try:
+        organization = get_organization_for_user(user=user)
+    except ObjectDoesNotExist:
+        return reverse(settings.LOGOUT_REDIRECT_URL)
+    site = get_site_by_organization(organization=organization)
+
+    return '{protocol}://{site_domain}'.format(
```

Comment:
I think this is missing `/logout` url, right?

```suggestion
    return '{protocol}://{site_domain}/logout'.format(
```

Reply:
actually no, should we logout LMS when Studio is logged out?

Reply:
@OmarIthawi  my thought is that signout from LMS doesn't touch Studio, so Studio shouldn't touch LMS, right?

Reply:
> signout from LMS doesn't touch Studio, 

Yes. That's good.

> so Studio shouldn't touch LMS, right?

No. I think we discussed to make Studio logout, also logout from LMS. Just to make it less confusing for course authors.

Reply:
Done, redirect with logout 👍🏼